File: lists_and_hashing/group_anagrams.py
from typing import List


class Solution:
    def group_anagrams(self, strs: List[str]) -> List[List[str]]:
        dictionary = {}
        for string in strs:
            sorted_string = ''.join(sorted(string))
            if sorted_string in dictionary:
                dictionary.get(sorted_string).append(string)
            else:
                dictionary[sorted_string] = [string]
        return dictionary.values()
    
my_solution = Solution()
strs = ["eat","tea","tan","ate","nat","bat"]
print(my_solution.group_anagrams(strs))

File: lists_and_hashing/groupAnagrams.js
class Solution {
  groupAnagrams(strs) {
    const dictionary = {};
    for (var i = 0; i < strs.length; i++) {
      let sorted_string = strs[i].split("").sort().join("");
      if (sorted_string in dictionary) {
        dictionary[sorted_string].push(strs[i]);
      } else {
        dictionary[sorted_string] = [strs[i]];
      }
    }
    return Object.values(dictionary);
  }
}

my_solution = new Solution();
strs = ["eat", "tea", "tan", "ate", "nat", "bat"];
console.log(my_solution.groupAnagrams(strs));
